### PR TITLE
Azure SQL Transparent Encryption: ensure proper result on exported resources

### LINKED ISSUE #2059: Transparent Data Encryption report failure on exported resources
**Description of the issue**

When doing an analysis of an existing sql server (exported from deployed resources), the TDE for Azure SQL keeps telling that servers are not compliant.

This is happening because the TDE check is done on the wrong propertie (for this particular export).
There is a check for the API version but I am guessing it is wrongly reporter by the Export job.

**To Reproduce**

Steps to reproduce the issue:
- export an existing compliant SQL (with TDE active) with Export-AzRuleData
- run azure check and see that the server is reported as failed

this is what was exported by the Export job
```json
 {
	"ResourceId": "/subscriptions/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/resourceGroups/MyResourceGroupRG/providers/Microsoft.Sql/servers/mysqlserver/databases/mysqldatabase/transparentDataEncryption/current",
	"Id": "/subscriptions/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/resourceGroups/MyResourceGroupRG/providers/Microsoft.Sql/servers/mysqlserver/databases/mysqldatabase/transparentDataEncryption/current",
	"Identity": null,
	"Kind": null,
	"Location": "West Europe",
	"ManagedBy": null,
	"ResourceName": "current",
	"Name": "current",
	"ExtensionResourceName": null,
	"ParentResource": null,
	"Plan": null,
	"Properties": {
	  "status": "Enabled"
	},
	"ResourceGroupName": "MyResourceGroupRG",
	"Type": "Microsoft.Sql/servers/databases/transparentDataEncryption",
	"ResourceType": "Microsoft.Sql/servers/databases/transparentDataEncryption",
	"ExtensionResourceType": null,
	"Sku": null,
	"Tags": null,
	"TagsTable": null,
	"SubscriptionId": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
	"CreatedTime": null,
	"ChangedTime": null,
	"ETag": null
  }
```

**Expected behaviour**

The output was expected to be true.

I have solved it by changing the rule in order to add a second check on the field
```powershell
# Synopsis: Enable transparent data encryption
Rule 'Org.SQL.TDE' -Type 'Microsoft.Sql/servers/databases', 'Microsoft.Sql/servers/databases/transparentDataEncryption' -If { !(IsMasterDatabase) } {
    $configs = @($TargetObject);
    if ($PSRule.TargetType -eq 'Microsoft.Sql/servers/databases') {
        $configs = @(GetSubResources -ResourceType 'Microsoft.Sql/servers/databases/transparentDataEncryption');
    }
    if ($configs.Length -eq 0) {
        if (!(IsExport)) {
            return $Assert.Pass();
        }
        return $Assert.Fail($LocalizedData.SubResourceNotFound, 'Microsoft.Sql/servers/databases/transparentDataEncryption');
    }
    foreach ($config in $configs) {
        if ($Assert.HasFieldValue($config, 'apiVersion', '2014-04-01').Result) {
            $Assert.HasFieldValue($config, 'Properties.status', 'Enabled');
        }
        elseif ($Assert.HasField($config, 'Properties.status').Result) {
            $Assert.HasFieldValue($config, 'Properties.status', 'Enabled');
        }
        else {
            $Assert.HasFieldValue($config, 'Properties.state', 'Enabled');
        }
    }
}
````

another simpler way could be to remove the API check and just look for the field
```powershell
# Synopsis: Enable transparent data encryption
Rule 'Org.SQL.TDE' -Type 'Microsoft.Sql/servers/databases', 'Microsoft.Sql/servers/databases/transparentDataEncryption' -If { !(IsMasterDatabase) } {
    $configs = @($TargetObject);
    if ($PSRule.TargetType -eq 'Microsoft.Sql/servers/databases') {
        $configs = @(GetSubResources -ResourceType 'Microsoft.Sql/servers/databases/transparentDataEncryption');
    }
    if ($configs.Length -eq 0) {
        if (!(IsExport)) {
            return $Assert.Pass();
        }
        return $Assert.Fail($LocalizedData.SubResourceNotFound, 'Microsoft.Sql/servers/databases/transparentDataEncryption');
    }
    foreach ($config in $configs) {
        if ($Assert.HasField($config, 'Properties.status').Result) {
            $Assert.HasFieldValue($config, 'Properties.status', 'Enabled');
        }
        else {
            $Assert.HasFieldValue($config, 'Properties.state', 'Enabled');
        }
    }
}
````

**Error output**
no error output

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **1.24.2**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.3.3
PSEdition                      Core
GitCommitId                    7.3.3
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Additional context**

I have made the change in a fork, I can merge any of the two solutions

### PR DESCRIPTION
**Pull request by zilberd**
## PR Summary

This fixes #2059 where exported resources are sometime flaged as failed with TDE.

I have simplified the check in order so it would verifiy the precense of the two potential fields.

## PR Checklist

- [X] PR has a meaningful title
- [X] Summarized changes
- [X] Change is not breaking
- [X] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [ ] Unit tests created/ updated
  - [ ] Rule documentation created/ updated
  - [X] Link to a filed issue
  - [X] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by zilberd** [comment]
@microsoft-github-policy-service agree

File: docs/CHANGELOG-v1.md [hunk 1]
```diff
@@ -36,6 +36,9 @@ What's changed since pre-release v1.25.0-B0035:
     [#2052](https://github.com/Azure/PSRule.Rules.Azure/pull/2052)
   - Bump Microsoft.NET.Test.Sdk to v17.5.0.
     [#2055](https://github.com/Azure/PSRule.Rules.Azure/pull/2055)
+- Bug fixes:
+  - SQL transparent data Encryption (TDE) works properly on all resources including exported resources
```
Comment (BernieWhite) [inline docs/CHANGELOG-v1.md:40]:
```suggestion
  - Fixed SQL transparent data Encryption (TDE) works properly on all resources including exported resources by @zilberd.
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
@zilberd Thanks for the PR. Looks great to me, just a tweak to the change log and we're good to go.

**Comment by zilberd** [comment]
thanks, I accepted the correction ;)

**Pull request merged**